hail/python/hailtop/aiocloud/aiogoogle/client/storage_client.py

```diff
@@ -639,22 +640,76 @@ def schemes() -> Set[str]:
     def storage_location(self, uri: str) -> str:
         return self.get_bucket_and_name(uri)[0]
 
-    async def is_hot_storage(self, location: str) -> bool:
+    async def is_hot_storage(self, uri: str) -> bool:
         """
         See `the GCS API docs https://cloud.google.com/storage/docs/storage-classes`_ for a list of possible storage
         classes.
 
         Raises
         ------
         :class:`aiohttp.ClientResponseError`
-            If the specified bucket does not exist, or if the account being used to access GCS does not have permission
-            to read the bucket's default storage policy.
+            If the specified object does not exist, or if the account being used to access GCS does not have permission
+            to read the bucket's default storage policy and it is not a public access bucket.
         """
-        return (await self._storage_client.bucket_info(location))["storageClass"].lower() in (
-            "standard",
-            "regional",
-            "multi_regional",
-        )
+        is_hot_storage = False
+        hot_storage_classes = {"standard", "regional", "multi_regional"}
+        errors: List[aiohttp.ClientResponseError] = []
+        location = self.storage_location(uri)
+
+        async def is_bucket_hot() -> bool:
+            try:
+                return (await self._storage_client.bucket_info(location))["storageClass"].lower() in hot_storage_classes
+            except aiohttp.ClientResponseError as e:
```

Comment:
Could you explain what kinds of `ClientResponseError`s you've been seeing here? I'd prefer to match on exactly the error that we expect regarding not being able to read a public bucket's storage class and otherwise fail fast, rather than accumulating these errors.

Reply:
went through and looked at the specific errors i'm getting; for `is_bucket_hot`, matched on the specific error message, and for the other two inner functions, realized the methods i'm calling ultimately give me `FileNotFoundError`s instead of `ClientResponseError`s, so matched on that error type instead!